Src/FluentAssertions/Types/TypeSelectorAssertions.cs

```diff
@@ -292,6 +292,131 @@ public AndConstraint<TypeSelectorAssertions> NotBeDecoratedWithOrInherit<TAttrib
             return new AndConstraint<TypeSelectorAssertions>(this);
         }
 
+        /// <summary>
+        /// Asserts that the current <see cref="Type"/> is in the specified <paramref name="namespace"/>.
+        /// </summary>
+        /// <param name="namespace">
+        /// The namespace that the type must be in.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<TypeSelectorAssertions> BeInNamespace(string @namespace, string because = "", params object[] becauseArgs)
+        {
+            Type[] typesNotInNamespace = Subject
+                .Where(t => t.Namespace != @namespace)
+                .ToArray();
+
+            Execute.Assertion
+                .ForCondition(!typesNotInNamespace.Any())
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected all types to be in namespace {0}{reason}," +
+                          " but the following types are in a different namespace:{1}{2}.",
+                    @namespace,
+                    Environment.NewLine,
+                    GetDescriptionsFor(typesNotInNamespace));
+
+            return new AndConstraint<TypeSelectorAssertions>(this);
+        }
+
+        /// <summary>
+        /// Asserts that the current <see cref="Type"/> is not in the specified <paramref name="namespace"/>.
+        /// </summary>
+        /// <param name="namespace">
+        /// The namespace that the type must not be in.
+        /// </param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <see cref="because" />.
+        /// </param>
+        public AndConstraint<TypeSelectorAssertions> NotBeInNamespace(string @namespace, string because = "", params object[] becauseArgs)
+        {
+            Type[] typesInNamespace = Subject
+                .Where(t => t.Namespace == @namespace)
+                .ToArray();
+
+            Execute.Assertion
+                .ForCondition(!typesInNamespace.Any())
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected all types to not be in namespace {0}{reason}," +
```

Comment:
_Expected none of the types to be in namespace_

Reply:
One could consider changing the wording in TypeAssertionSpecs as well, for example at https://github.com/fluentassertions/fluentassertions/blob/767a86bbf74ba5894102e24b63dbc004fc0d308a/Tests/Shared.Specs/TypeAssertionSpecs.cs#L1254